clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderFollowerStoreIngestionTask.java

```diff
@@ -1066,17 +1066,30 @@ protected void syncConsumedUpstreamRTOffsetMapIfNeeded(
       Map<String, PubSubPosition> upstreamStartPositionByPubSubUrl) {
     // Update in-memory consumedUpstreamRTOffsetMap in case no RT record is consumed after the subscription
     final PubSubTopic leaderTopic = pcs.getOffsetRecord().getLeaderTopic(pubSubTopicRepository);
+    final PubSubTopicPartition leaderTopicPartition = pcs.getSourceTopicPartition(leaderTopic);
     if (leaderTopic != null && leaderTopic.isRealTime()) {
       upstreamStartPositionByPubSubUrl.forEach((kafkaURL, upstreamStartPosition) -> {
-        if (upstreamStartPosition
-            .getNumericOffset() > getLatestConsumedUpstreamPositionForHybridOffsetLagMeasurement(pcs, kafkaURL)
-                .getNumericOffset()) {
+        PubSubPosition latestConsumedRtPosition =
+            getLatestConsumedUpstreamPositionForHybridOffsetLagMeasurement(pcs, kafkaURL);
+        // Only update if the new position is earlier than the existing one
```

Comment:
This comment is incorrect:
It should be update latest consumed if incoming upstream start position is greater than the latest consumed. 

Also IMO we should probably skip updating latest consumed RT position if `latestConsumedRtPosition` is `EARLIEST`